public/app/features/trails/DataTrail.tsx

```diff
@@ -208,12 +287,321 @@ export class DataTrail extends SceneObjectBase<DataTrailState> {
+      const filtersVariable = sceneGraph.lookupVariable(VAR_FILTERS, this);
+
+      const datasourceUid = sceneGraph.interpolate(trail, VAR_DATASOURCE_EXPR);
+
+      const otelTargets = await totalOtelResources(datasourceUid, timeRange);
+      const deploymentEnvironments = await getDeploymentEnvironments(datasourceUid, timeRange);
+      const hasOtelResources = otelTargets.job !== '' && otelTargets.instance !== '';
+      // const isStandard = await isOtelStandardization(datasourceUid, timeRange);
+      if (
+        otelResourcesVariable instanceof AdHocFiltersVariable &&
+        otelDepEnvVariable instanceof CustomVariable &&
+        otelJoinQueryVariable instanceof ConstantVariable &&
+        filtersVariable instanceof AdHocFiltersVariable
+      ) {
+        // HERE WE START THE OTEL EXPERIENCE ENGINE
+        // 1. Set deployment variable values
+        // 2. update all other variables and state
+        // Note** isStandard is not reliable because of an issue where there are
+        // multiple target_info series with job&instance pairs due to staleness which breaks the join.
+        // We handle this by now using topk to choose 1 series to allow for the join
+        if (hasOtelResources && deploymentEnvironments.length > 0 /*&& isStandard*/) {
+          // *** apply VAR FILTERS manually
+          // otherwise they will appear anywhere the query contains {} characters
+          filtersVariable.setState({
+            addFilterButtonText: 'Select metric attributes',
+            label: 'Select metric attribute',
+          });
+
+          // 1. set deployment variable values
+          let varQuery = '';
+          const options = deploymentEnvironments.map((env) => {
+            varQuery += env + ',';
+            return { value: env, label: env };
+          });
+          // We have to have a default value because custom variable requires it
+          // we choose one default value to help filter metrics
+          // The work flow for OTel begins with users selecting a deployment environment
+          const defaultDepEnv = options[0].value; // usually production
+          // On starting the explore metrics workflow, the custom variable has no value
+          // Even if there is state, the value is always ''
+          // The only reference to state values are in the text
+          const otelDepEnvValue = otelDepEnvVariable.state.text;
+
+          // TypeScript issue: VariableValue is either a string or array but does not have any string or array methods on it to check that it is empty
+          const notInitialvalue = otelDepEnvValue !== '' && otelDepEnvValue.toLocaleString() !== '';
+
+          const depEnvInitialValue = notInitialvalue ? otelDepEnvValue : defaultDepEnv;
+
+          otelDepEnvVariable?.setState({
+            // cannot have an undefined custom value
+            // this breaks everything
+            // create an issue for this
```

Comment:
Did we create an issue for this? If yes could you please link it here?

Reply:
This issue has been fixed! I am taking the comments out.